File: Examples/R/tests/testSpatial.R
test_that("Spatial exports construct and preserve coordinates", {
  for (className in c("FMTcoordinate", "FMTforest", "FMTSpatialSchedule")) {
    expect_true(exists(className, inherits = TRUE))
  }

  coordinate <- new(FMTcoordinate, 4L, 7L)
  expect_equal(coordinate$getx(), 4L)
  expect_equal(coordinate$gety(), 7L)
})

File: Examples/Python/tests/testSpatial.py
from FMT import Spatial


def testCoordinateRoundTrip():
    coordinate = Spatial.FMTcoordinate(4, 7)
    assert coordinate.getx() == 4
    assert coordinate.gety() == 7


def testSpatialObjectsCanBeConstructed():
    for name, constructor in {
        "FMTforest": Spatial.FMTforest,
        "FMTeventcontainer": Spatial.FMTeventcontainer,
        "FMTSpatialSchedule": Spatial.FMTSpatialSchedule,
    }.items():
        assert constructor() is not None
